ibm/service/cos/resource_ibm_cos_backup_policy_test.go

```diff
@@ -0,0 +1,922 @@
+	policyName2 := fmt.Sprintf("backup-policy2%d", acctest.RandIntRange(10, 100))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMCosBucketDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMCosBackup_Policy_Multiple_Policy_Valid(acc.CosCRN, bucketName, backupVaultName1, backupVaultName2, region, accountID, policyName1, policyName2, guid),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("ibm_cos_backup_vault.backup-vault1", "backup_vault_name", backupVaultName1),
+					resource.TestCheckResourceAttr("ibm_cos_backup_vault.backup-vault2", "backup_vault_name", backupVaultName2),
+					resource.TestCheckResourceAttr("ibm_cos_backup_policy.policy1", "policy_name", policyName1),
+					resource.TestCheckResourceAttr("ibm_cos_backup_policy.policy2", "policy_name", policyName2),
+				),
+			},
+		},
+	})
+}
+
+func TestAccIBMCosBackup_Policy_Non_Versioning_Source_Bucket_Invalid(t *testing.T) {
+	accountID := acc.IBM_AccountID_REPL
+	bucketName := fmt.Sprintf("terraform-backup-source%d", acctest.RandIntRange(10, 100))
+	backupVaultName := fmt.Sprintf("terraform-backup-vault%d", acctest.RandIntRange(10, 100))
+	region := "us"
+	guid := strings.Split(acc.CosCRN, ":")[7]
+	policyName := fmt.Sprintf("backup-policy%d", acctest.RandIntRange(10, 100))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMCosBucketDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCheckIBMCosBackup_Policy_Non_Versioning_Source_Bucket_Invalid(acc.CosCRN, bucketName, backupVaultName, region, accountID, policyName, guid),
+				ExpectError: regexp.MustCompile("Source bucket does not have versioning enabled"),
+			},
+		},
+	})
+}
+
+func TestAccIBMCosBackup_Policy_Multiple_Invalid(t *testing.T) {
+	accountID := acc.IBM_AccountID_REPL
+	bucketName := fmt.Sprintf("terraform-backup-source%d", acctest.RandIntRange(10, 100))
+	backupVaultName := fmt.Sprintf("terraform-backup-vault%d", acctest.RandIntRange(10, 100))
+	region := "us"
+	guid := strings.Split(acc.CosCRN, ":")[7]
+	policyName1 := fmt.Sprintf("backup-policy1%d", acctest.RandIntRange(10, 100))
+	policyName2 := fmt.Sprintf("backup-policy2%d", acctest.RandIntRange(10, 100))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMCosBucketDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCheckIBMCosBackup_Policy_Multiple_Invalid(acc.CosCRN, bucketName, backupVaultName, region, accountID, policyName1, policyName2, guid),
+				ExpectError: regexp.MustCompile("Bad Request"),
+			},
+		},
+	})
+}
+
+func TestAccIBMCosBackup_Policy_Invalid_Backup_Type(t *testing.T) {
```

Comment:
you defined invalid backup type, but no parameter related to that?

Reply:
I added the invalid backup type directly in the 
`testAccCheckIBMCosBackup_Policy_Invalid_Backup_Type`
function